lib/timber/events/custom_event.ex

```diff
@@ -6,22 +6,74 @@ defmodule Timber.Events.CustomEvent do
   to your line of business like receiving credit card payments, adding products
   to a card, saving a draft of a post, or changing a user's password.
 
-  Custom events take a `name` and a map of `data`. You can choose either strings
-  or atoms as keys, and values can contain nested maps. The only requirement is
-  that is that you don't use tuples as this will cause an encoder issue.
+  Event data can include anything that adheres to the `Poison.Encoder` protocol.
+  That is, anything that can be encoded to JSON, including nested maps.
 
-  The resulting event can be passed to `Logger` in the `:timber_event` key of
-  the metadata. See the documentation for `new/1`.
+  ## Examples
+
+  Basic example:
+
+    iex> require Logger
+    iex> event_data = %{customer_id: "xiaus1934", amount: 1900, currency: "USD"}
+    iex> event = Timber.event(name: :payment_received, data: event_data)
+    iex> Logger.info("Received payment", timber_event: event)
+
+  With timing: (automatically calculate the time_ms attribute)
+
+    iex> require Logger
+    iex> timer = Timber.start_timing()
+    iex> # ... code to time ...
+    iex> event_data = %{customer_id: "xiaus1934", amount: 1900, currency: "USD"}
+    iex> event = Timber.event(name: :payment_received, data: event_data, timer: timer)
+    iex> Logger.info("Received payment", timber_event: event)
+
+  A more advanced example:
+
+  At Timber we like to define and structure our events, similar to how we define exceptions.
+  As a result, you can also do:
+
+    iex> require Logger
+    iex> defmodule PaymentReceivedEvent do
+    iex>   use Timber.Events.CustomEvent
+    iex>   defstruct [:customer_id, :amount, :currency]
+    iex> end
+    iex> timer = Timber.start_timing()
+    iex> # ... code to time ...
+    iex> event = PaymentReceivedEvent.new(customer_id: "xiaus1934", amount: 1900, currency: "USD", timer: timer)
+    iex> Logger.info("Received payment", timber_event: event)
+
+  In the above example, the struct name, `PaymentReceivedEvent`, will be used as the custom event
+  name.
   """
 
   alias Timber.Timer
 
   @type t :: %__MODULE__{
     name: String.t,
-    data: map | nil,
+    data: map() | nil,
     time_ms: float() | nil
   }
 
+  @doc false
+  defmacro __using__(opts) do
+    quote bind_quoted: [opts: opts] do
```

Comment:
I don't know that we want to use `bind_quoted`. I'm pretty sure we just want to ignore `opts` in the `__using__` declaration (_i.e._, `__using__(_opts)`)